.storybook/polaris-readme-loader.js

```diff
@@ -187,52 +141,53 @@ function parseCodeExamples(data, file) {
     .split('###');
 
   const [, ...allExamples] = examplesAndHeader;
-  const filePath = file.split('polaris-react/').slice(-1)[0];
 
   if (allExamples.length === 0) {
-    throw new Error(
-      chalk`🚨 {red [${matter.data.name}]} No examples found in ${filePath}`,
-    );
+    console.log(chalk`🚨 {red [${
+      matter.data.name
+    }]} No examples found. For troubleshooting advice see https://github.com/Shopify/polaris-react/blob/master/documentation/Component%20READMEs.md#troubleshooting',
+    `);
   }
 
-  const webExamples = allExamples.filter((example) =>
-    isExampleForPlatform(example, 'web'),
-  );
+  const nameRegex = /(.)*/;
+  const codeRegex = /```jsx(.|\n)*?```/g;
 
-  const examples = webExamples.map((example) => {
-    const nameMatches = example.match(/(.)*/);
-    const codeBlock = example.match(/```jsx(.|\n)*?```/g);
+  const examples = allExamples
+    .filter((example) => isExampleForPlatform(example, 'web'))
+    .map((example) => {
+      const nameMatches = example.match(nameRegex);
+      const codeBlock = example.match(codeRegex);
 
-    const name = nameMatches !== null ? nameMatches[0].trim() : '';
-    const code =
-      codeBlock !== null ? wrapExample(stripCodeBlock(codeBlock[0])) : '';
+      const name = nameMatches !== null ? nameMatches[0].trim() : '';
+      const code =
+        codeBlock !== null ? wrapExample(stripCodeBlock(codeBlock[0])) : '';
 
-    return {name, slug: slugify(name), code};
-  });
+      // TODO need to strip out android/ios examples
+      const description = example
+        .replace(nameRegex, '')
+        .replace(codeRegex, '')
+        .trim();
+
+      return {name, slug: slugify(name), code, description};
+    });
 
   if (examples.filter((example) => example.code).length === 0) {
-    throw new Error(
-      chalk`🚨 {red [${
-        matter.data.name
-      }]} At least one react example expected in ${filePath}`,
+    console.log(
+      chalk`🚨 {red [${matter.data.name}]} At least one react example expected`,
     );
   }
 
   examples.forEach((example) => {
     if (example.code === '') {
-      throw new Error(
+      console.log(
         chalk`🚨 {red [${matter.data.name}]} Example “${
           example.name
-        }” is missing a React example in ${filePath}`,
+        }” is missing a React example`,
```

Comment:
It'd be useful to keep the file path - unless there's a technical reason why we're losing it? The reason is that you can cmd+click the file path in the terminal to access the file quicker than having to go back to your text editor.

Reply:
Webpack loaders act upon a string of text, not a file (this is what allows you to string multiple loaders together one after another). Because of that we don't have access to the file path.

Reply:
This 'worked' in the past because the old loader wasn't actually using the source content that webpack gave it and instead read files directly. This new way is more in line with how webpack loaders should work

Reply:
Thanks, no worries then!